tools/asset-sync/assets.Tests.ps1

```diff
@@ -0,0 +1,430 @@
+    }
+
+    It "Should should resolve a custom repoId." {
+      $jsonContent = Get-Basic-AssetsJson
+      $jsonContent.AssetsRepoId = "custom"
+
+      $testPath = Describe-TestFolder -AssetsJsonContent $jsonContent -Files @()
+      $config = Resolve-AssetsJson -TargetPath $testPath
+      $result = Resolve-AssetRepo-Location -Config $config
+      $expectedHash = "D41D8CD98F00B204E9800998ECF8427E"
+
+      $expectedLocation = Resolve-Path (Join-Path $PSScriptRoot ".." ".." ".assets" "$expectedHash")
+      
+      $result | Should -Be $expectedLocation.ToString()
+    }
+  }
+
+  Context "Update-AssetsJson" {
+    It "Should update a targeted recording.json w/ a new SHA and output without mangling the json file." {
+      $testLocation = Describe-TestFolder -AssetsJsonContent (Get-Basic-AssetsJson) -Files @()
+      $config = Resolve-AssetsJson -TargetPath $testLocation
+
+      $config.AssetsJsonLocation
+    }
+
+    It "Should no-op if no change." {
+      $testLocation = Describe-TestFolder -AssetsJsonContent (Get-Basic-AssetsJson) -Files @()
+      $config = Resolve-AssetsJson -TargetPath $testLocation
+
+      $config.AssetsJsonLocation
+    }
+  }
+
+  Context "Resolve-CheckoutPaths" {
+    It "Should correctly resolve a non-root checkout path format." {
+      $files = @(
+        "sdk/storage/",
+        "sdk/storage/assets.json",
+        "sdk/storage/azure-storage-blob/awesome.json"
+      )
+      $assetsContent = Get-Basic-AssetsJson
+      $testLocation = Describe-TestFolder -AssetsJsonContent $assetsContent -Files $files
+      $config = Resolve-AssetsJson -TargetPath (Join-Path $testLocation "sdk" "storage")
+
+      $checkoutPaths = Resolve-CheckoutPaths -Config $config
+      $checkoutPaths | Should -Be (Join-Path "recordings" "sdk" "storage").Replace("`\", "/")
+    }
+
+    It "Should correctly resolve a root checkout path." {
+      $files = @(
+        "sdk/storage/",
+        "sdk/storage/azure-storage-blob/awesome.json"
+      )
+      $assetsContent = Get-Basic-AssetsJson
+      $testLocation = Describe-TestFolder -AssetsJsonContent $assetsContent -Files $files
+      $config = Resolve-AssetsJson -TargetPath (Join-Path $testLocation "sdk" "storage")
+
+      $checkoutPaths = Resolve-CheckoutPaths -Config $config
+      $checkoutPaths | Should -Be "recordings/"
+    }
+  }
+
+  Context "Initialize-AssetsRepo" {
+    It "Should create assets repo for standard sync." {
+      $files = @(
+        "sdk/storage/",
+        "sdk/storage/azure-storage-blob/awesome.json"
+      )
+      $assetsContent = Get-Basic-AssetsJson
+      $testLocation = Describe-TestFolder -AssetsJsonContent $assetsContent -Files $files
+
+      $config = Resolve-AssetsJson -TargetPath $testLocation
+
+      Initialize-AssetsRepo -Config $config
+      $assetLocation = Resolve-AssetRepo-Location -Config $config
+
+      $result = Is-AssetsRepo-Initialized -Config $config
+
+      $result | Should -Be "$true"
```

Comment:
Are you expecting result to be a boolean or a stringified powershell boolean or a "true" string? Seems like powershell kind of treats this string as equivalent to a boolean which makes it more confusing.

Reply:
Oh. I don't know why I made it a stringified thing. It should be a boolean tbh. It's just that Pshell is doing casting of both auto-magically and isn't erroring.

(This is with strict mode enabled btw *facepalm*)